### DISCUSSION TITLE
How do I change my domain name?

### DISCUSSION THREAD

**Discussion by Mr-Ao-Dragon** [serverless-dns/serverless-dns#207] [Q&A] [answered]
In some countries, worker.dev is blocked and you will not be able to use the service without using your own domain name, seek a solution here.

**Comment by abewartech** [comment, marked as answer]
You're correct. In some countries, *.worker.dev domains are blocked, making the default Serverless DNS service inaccessible.

The solution is to use your own custom domain to access your Serverless DNS instance. This will make your DNS queries appear to go to your domain instead of worker.dev.

Prerequisites:

1. Your own domain (e.g., mydomain.com)

2. A Cloudflare account with your domain added and managed by Cloudflare (using Cloudflare’s nameservers).


Step-by-Step Guide
Step 1: Deploy the Serverless DNS Worker

1. Fork/clone the [Serverless DNS repository](https://github.com/serverless-dns/serverless-dns).
2. Follow the deployment instructions in the README to deploy the Worker to your own Cloudflare account.
3. Once deployed, note your Worker’s default URL (e.g., https://serverless-dns.your-account.workers.dev).



Step 2: Add a Custom Domain in Cloudflare

1. Log into the Cloudflare dashboard and select your domain (e.g., mydomain.com).
2. Go to Workers & Pages > Overview.
3. Select your deployed Worker (e.g., serverless-dns).
4. Go to the Triggers tab.
5. Under Custom Domains, click Add Custom Domain.
6. Enter the subdomain you want to use for DNS.
7. Example: dns.mydomain.com or doh.mydomain.com
8. Cloudflare will automatically create the necessary DNS record (a CNAME or _worker.domain route) for you.


Step 3: Configure Your Devices

1. Update your DNS settings on your device, router, or in your DNS client (like Intra, RethinkDNS, or personal DNS filter) to use your new custom domain.

- Plain DNS: Use dns.mydomain.com (if you configured it to listen on port 53 via a proxy, though Workers only support HTTP/HTTPS directly).
- DNS-over-HTTPS (DoH): Use https://doh.mydomain.com/dns-query
- DNS-over-TLS (DoT): Requires additional setup (like a proxy), as Workers don't natively support DoT.
